### PR TITLE
fix event duplication, add example, add tests

### PR DESCRIPTION
**Pull request by aggyomfg**
Hello, thank you for this lib, it's very useful.
But looks like I found a bug in event handling.

I wrote 'in_range.rs' example, and it's not working as I expected.

With Poly-non-symetric Relation:
```rust 
#[derive(Relation, Component)]
#[aery(Poly)]
struct InRange;
```

We can see 2 SetEvent and 4 UnsetEvent triggers, but I expected 2 SetEvent and 2 UnsetEvent:
```log 
2024-12-01T08:02:09.898181Z  INFO in_range: 3v1 is now in range of 4v1
2024-12-01T08:02:09.898416Z  INFO in_range: 4v1 is now in range of 3v1
2024-12-01T08:02:11.928483Z  INFO in_range: 4v1 is no longer in range of 3v1
2024-12-01T08:02:11.928670Z  INFO in_range: 3v1 is no longer in range of 4v1
2024-12-01T08:02:11.928774Z  INFO in_range: 3v1 is no longer in range of 4v1
2024-12-01T08:02:11.928861Z  INFO in_range: 3v1 is no longer in range of 4v1
```

If I add Symetric to relation:
```rust
#[derive(Relation, Component)]
#[aery(Poly, Symmetric)]
struct InRange;
```

Situation become even worse:
```log
2024-12-01T08:07:19.816287Z  INFO in_range: 3v1 is now in range of 4v1
2024-12-01T08:07:19.816422Z  INFO in_range: 4v1 is now in range of 3v1
2024-12-01T08:07:19.816473Z  INFO in_range: 4v1 is now in range of 3v1
2024-12-01T08:07:19.816508Z  INFO in_range: 3v1 is now in range of 4v1
2024-12-01T08:07:21.917999Z  INFO in_range: 4v1 is no longer in range of 3v1
2024-12-01T08:07:21.918128Z  INFO in_range: 3v1 is no longer in range of 4v1
2024-12-01T08:07:21.918185Z  INFO in_range: 3v1 is no longer in range of 4v1
2024-12-01T08:07:21.918223Z  INFO in_range: 3v1 is no longer in range of 4v1
2024-12-01T08:07:21.918260Z  INFO in_range: 3v1 is no longer in range of 4v1
2024-12-01T08:07:21.918296Z  INFO in_range: 4v1 is no longer in range of 3v1
```

And after fix it looks like that:
```log
2024-12-01T08:11:03.409970Z  INFO in_range: 3v1 is now in range of 4v1
2024-12-01T08:11:03.410154Z  INFO in_range: 4v1 is now in range of 3v1
2024-12-01T08:11:05.582783Z  INFO in_range: 4v1 is no longer in range of 3v1
2024-12-01T08:11:05.582881Z  INFO in_range: 3v1 is no longer in range of 4v1
```

Also added some tests to check this events behavior.
Please check examples and test to understand situation.

Hope together we can discuss and solve it, if I got something wrong.

### CODE REVIEW

**Comment by iiYese** [comment]
Looks correct. I can review & merge when you're ready to take it out of draft.

**Comment by aggyomfg** [comment]
> Looks correct. I can review & merge when you're ready to take it out of draft.
Thanks.
I've identified some additional cases that might cause unexpected issues with non-sending. However, I don't have time to investigate them further right now. That's the reason of draft, I'll look into this later and update you.

**Comment by aggyomfg** [comment]
@iiYese hey, I think It's ready for review now, i added additional target_has_host_as_target check in UnsetAsymmetric and wrote better tests for events.

File: examples/in_range.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+#[derive(Component)]
+struct MovingEntity {
+    direction: f32, // Movement direction along the x-axis, positive or negative.
+}
+
+#[derive(Component)]
+struct Name(&'static str);
+
+impl Display for Name {
```
Comment (iiYese) [inline examples/in_range.rs:15]: I'd just slap a `Deref` derive on the component or use `.0` instead when accessing. No need to noisy up examples with boilerplate.

**Review by iiYese — COMMENTED** [review]
Looks good just needs a few minor changes.

File: examples/in_range.rs (hunk 1 above)
Comment (aggyomfg) [reply examples/in_range.rs]: reworked example to be more readable and less noisy

File: src/edges.rs [hunk 2]
```diff
@@ -433,7 +436,7 @@ where
 
         // Symmetric set has to happen before exclusivity unset otherwise
         // an entity can get despawned when it shouldn't.
-        if R::SYMMETRIC && !self.symmetric_action {
+        if (R::SYMMETRIC && R::EXCLUSIVE) && !self.symmetric_action {
```
[the remark this answers is not in the document]
Comment (aggyomfg) [reply src/edges.rs]: you right, reworked set command to check if targets/hosts already persist before adding.

File: examples/in_range.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.0)
+    }
+}
+
+// Define a relationship where multiple entities can be linked non-exclusively.
+// Symmetry is not set in this example, it may arise from the implementation,
+// when we set Poly relation from both sides.
+// But adding the aery `Symmetric` attribute to the struct would enforce it, and MUST show same results.
```
[the remark this answers is not in the document]
Comment (aggyomfg) [reply examples/in_range.rs]: removed

**Comment by aggyomfg** [comment]
@iiYese thanks, fixed review comments, and made Set command more safe (check if already exists)

**Review by iiYese — APPROVED** [review]

**Pull request merged**